src/state-manager.ts

```diff
@@ -0,0 +1,85 @@
+import * as vscode from "vscode";
+import { Environment } from "./env0-environments-provider";
+
+export class StateManager {
```

Comment:
BAD NAME, when ever you have "manager" in the name should rise a warning.
maybe
`extension-state`
`extension-state-service`

Reply:
43cf3f6787a0d6fc61607d41cb77a7fffff10a3b